_drafts/oryx/orchestration/app-plugins.md

```diff
@@ -0,0 +1,135 @@
+
+There are several built-in plugins provided by the Oryx framework:
+
+- [`ComponentsPlugin`](#components-plugin)
+- [`InjectionPlugin`](#injection-plugin)
+- [`ResourcePlugin`](#resource-plugin)
+- [`ThemePlugin`](#theme-plugin)
+
+For ordinary application development, there's no need to develop custom plugins.
+
+## Built-in plugins
+
+### Components plugin
+
+The `ComponentsPlugin` is an important plugin that orchestrates the (lazy) loading of components. The plugin registers all the component definitions and loads the implementation whenever required in the DOM.
+
+The plugin exposes the following API:
+
+- `registerComponents()` – registers components
+
+For example, to register more component:
+
+```ts
+import { ComponentsPlugin } from '@spryker-oryx/core';
+
+app.requirePlugin(ComponentsPlugin).registerComponents([
+  ... // <-- Component definitions go here
+])
+```
+
+### Injection plugin
+
+The `InjectionPlugin` plugin manages the dependency injection system (DI). DI is an important concept in Oryx that allows to customise application logic which it nested deep inside the application logic.
+
+The plugin exposes the following API:
+
+- `getInjector()` – returns the injector that it has created
+- `createInjector()` – recreates the injector with the configured providers
+
+For example, to get injector and inject something from it:
+
+```ts
+import { InjectionPlugin } from '@spryker-oryx/core';
+
+app.requirePlugin(InjectionPlugin).inject(...)
+```
+
+The plugin will expose the `App` instance as an `AppRef` token, so that the main application reference is available throughout the application:
+
+```ts
+import { AppRef } from "@spryker-oryx/core";
+import { inject } from "@spryker-oryx/di";
+
+class MyService {
+  constructor(private app = inject(AppRef)) {
+    // Use `app` here
+  }
+}
+```
+
+### Resource plugin
+
+This plugin adds support for the application resources, such as images. The ResourcesPlugin lazy loads resources into the application whenever they're needed.
+
+Whenever resources are added to the Oryx application this plugin will be automatically configured and used.
+
+The plugin exposes the following API:
+
+- `getResources()` – gets all registered resources
+- `getGraphicValue()` – Loads graphical resources
+
+For example, to get all resources:
+
+```ts
+import { ResourcePlugin } from "@spryker-oryx/core";
+
+app.requirePlugin(ResourcePlugin).getResources();
+```
+
+### Theme plugin
+
+This plugin adds support for the Application themes. Whenever a theme is added to the Oryx application this plugin will be automatically configured and used.
+
+The plugin exposes the following API:
+
+- Get configured icons (`getIcons`)
+- Get specific icon (`getIcon`)
+- Get configured breakpoints (`getBreakpoints`)
+- Resolve component theme (`resolve`)
+- Normalize styles (`normalizeStyles`)
+- Interpolate media queries (`generateMedia`)
+
+For example, to get all icons:
```

Comment:
In the list above, there is no API that gets _all_ icons. Please clarify

Reply:
Actually `Get configured icons` means get all icons =)

Reply:
1. Is it important to identify that they are _configured_?
2. Which of the items in the list are configured?

Reply:
1. Yes since they are coming from a theme configuration
2. Every method returns stuff that is configured in a theme that's used in the app